src/types.h

```diff
@@ -412,11 +412,11 @@ constexpr Rank rank_of(Square s) {
 }
 
 constexpr Square relative_square(Color c, Square s) {
-  return Square(s ^ (c * 56));
+  return c == WHITE ? s : flip_rank(s);
```

Comment:
This and line 419 changes are actually somewhat slower on my machine.   It seems that introducing branching is not a good idea.

Reply:
I get bit by this occasionally.  Patches test faster on my machine, but others say it's slower.  Any ways to counteract that?  How are you testing performance?